_posts/2020-06-07-Human-Curation-and-Digital-Datasets.md

```diff
@@ -0,0 +1,213 @@
+as the latest version of the dataset is the format checking.  TRiG is
+very specific file format and it can be very easy for a human to break
+it.  In this case, a TRiG parser is used and any errors in file format
+is indicated to the curator for fixing.  While this does not always
+guarantee the files are not broken, it does give one last check before
+the data is merged.
+
+## Peer Review
+
+Another method used is peer review.  Using the Github Pull Request
+(sometimes also called Merge Request), changes to the data can be
+reviewed before it is merged into the data set as the current version.
+The pre-modern scribes and redactors also took a peer review approach
+as can be seen in whole areas of manuscripts being scraped and
+rewritten or comments put either interlinearly or on the sides of
+columns in a manuscript.  It works similarly where merge requests are
+given to a reviewer who goes over the information and approves it
+before it becomes part of the current dataset.  As this is a human
+process, albeit one assisted by automation, errors still slip through.
+
+Again, this process is constrained by the curator time available.  For
+IrishGen to move forward and avoid stagnation, lack of peer review
+cannot be a barrier to continued development.  Thus practical
+flexibility is necessary.  If there is no peer reviewer available then
+the head curator can just merge changes into the dataset without
+review.  This can also happen when another curator is working and the
+head curator is not available.  Additionally, as the dataset is
+publicly available,
+[forking](https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project)
+is a possibility to change the dataset.
+
+While peer review is a time honoured method to attempt to stop errors
+from entering a collaborative work, like the early Irish genealogies
+historically and IrishGen now, it cannot entirely stop errors from
+curators being introduced.
+
+## Audit
+
+This particular technique does not catch errors but detects them after
+they have become part of the dataset.  One way of doing this was
+recently undertaken by Eystein Thanisch who used the GraphDB
+triplestore which allows for storing, searching, and visualising RDF.
+This works by using the
+[SPARQL](https://www.w3.org/TR/sparql11-overview/) query language and
+Graph DB's visualisation to allow visual inspection of the data.  For
+instance, looking at part of Flaithbertach's genealogy from above, it
+may be visualised thus:
+
+<img src="{{site.baseurl}}/assets/images/HumanCuration_1.png" />
+
+However, due to the structure of the data, errors can be detected more
+easily if the visualisation does not make sense.  Many times this can
+be identified as a cyclical structure in the descendants of an
+individual where one of the descendants of an individual ends up being
+both an indirect and direct descendant of an individual who should be
+farther up the genealogical structure.  This visual inspection is the
```

Comment:
"further up the genealogical structure, or even their own ancestor." ?

Reply:
Maybe we should call this the "Philip J. Fry syndrome"... :P

Reply:
Just had to read up on Philip J. Fry, but, yes, I now see how that is apposite :-D